pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerBase.java

```diff
@@ -616,6 +642,25 @@ public CompletableFuture<Void> acknowledgeCumulativeAsync(MessageId messageId, T
         return doAcknowledgeWithTxn(messageId, AckType.Cumulative, Collections.emptyMap(), txnImpl);
     }
 
+    protected CompletableFuture<Void> acknowledgeCumulativeAsync(Map<String, MessageId> topicToMessageIdMap) throws PulsarClientException {
+        return acknowledgeCumulativeAsync(topicToMessageIdMap, null);
+    }
+
+    protected CompletableFuture<Void> acknowledgeCumulativeAsync(Map<String, MessageId> topicToMessageIdMap, Transaction txn) throws PulsarClientException {
```

Comment:
It looks like the keys are useless.
So why do we need this API?

Reply:
The base idea is to support cumulative acknowledgement over list of messages. But using List is confusing for the users. This Map directs them to maintain latest message for each partition and use it correctly. 
Can you please go through this email thread sent to _dev@pulsar.apache.org_ for full motivation and opinion of others
**[DISCUSS] User-friendly acknowledgeCumulative API on a partitioned topic or multi-topics**